profiling/src/lib.rs

```diff
@@ -160,13 +167,33 @@ pub extern "C" fn get_module() -> &'static mut zend::ModuleEntry {
         version: PROFILER_VERSION.as_ptr(),
         post_deactivate_func: Some(prshutdown),
         deps: DEPS.as_ptr(),
+        #[cfg(php_zts)]
+        globals_ctor: Some(ginit),
+        #[cfg(php_zts)]
+        globals_dtor: Some(gshutdown),
+        #[cfg(php_zts)]
+        globals_size: 1,
+        #[cfg(php_zts)]
+        globals_id_ptr: unsafe { &mut GLOBALS_ID_PTR },
```

Comment:
Do the C macros expand to having a global with `i32` and this mimics that, or is this a sort of hack?

Reply:
I documented the TSRM things that are happening there a bit more. It is actually a `ts_rsrc_id`, which is [defined to be a c_int](https://heap.space/xref/PHP-8.3/TSRM/TSRM.h?r=644d3628#60). Upon module startup, [`ts_allocate_id()` will be called](https://heap.space/xref/PHP-8.3/Zend/zend_API.c?r=d41e97ae#2303) and store the thread-safe-resource id there.